src/views/Setup.vue

```diff
@@ -1,27 +1,83 @@
 <script setup>
-import { ref, onMounted } from 'vue';
+import { ref, watch, onMounted, onUnmounted } from 'vue';
 import { useRouter } from 'vue-router';
+import { useWeb3 } from '@/composables/useWeb3';
+import { useModal } from '@/composables/useModal';
 import { setPageTitle } from '@/helpers/utils';
+import RegisterENS from '@/components/RegisterENS.vue';
+import { useApolloQuery } from '@/composables/useApolloQuery';
+import { ENS_QUERY } from '@/helpers/queries';
 
 const router = useRouter();
-const tlds = ['.eth', '.xyz', '.com', '.org', '.io', '.app', '.art'];
+const { web3, web3Account } = useWeb3();
+const { ensApolloQuery } = useApolloQuery();
+const { modalAccountOpen } = useModal();
 
-const id = ref('');
+onMounted(() => {
+  setPageTitle('page.title.setup');
+});
 
-function isValidTLD(id) {
-  return tlds.some(tlds => id.endsWith(tlds));
-}
+const validTlds = ['eth', 'xyz', 'com', 'org', 'io', 'app', 'art'];
+const ownedEnsDomains = ref([]);
 
-function handleSubmit() {
-  router.push({
-    name: 'spaceSettings',
-    params: { key: id.value.toLowerCase() }
-  });
-}
+const loadingOwnedEnsDomains = ref(false);
+const loadOwnedEnsDomains = async () => {
+  if (web3Account.value) {
+    loadingOwnedEnsDomains.value = true;
+    const res = await ensApolloQuery({
+      query: ENS_QUERY,
+      variables: {
+        id: web3Account.value.toLowerCase()
+      }
+    });
+    ownedEnsDomains.value = res.account?.domains || [];
+    loadingOwnedEnsDomains.value = false;
+  } else {
+    ownedEnsDomains.value = [];
+  }
+};
 
-onMounted(() => {
-  setPageTitle('page.title.setup');
+// used either on click on existing owned domain OR once a newly registered
+// domain is returned by the ENS subgraph.
+const goToSettings = key => {
+  router.push({ name: 'spaceSettings', params: { key } });
+};
+
+// input for new domain registration
+const newDomain = ref('');
+
+// indicates whether to periodically check for new domains
+// (after clicking on "Register")
+const waitingForRegistration = ref(false);
+// handle periodic lookup (every 10s) while registering new domain
+let waitingForRegistrationInterval;
+const waitForRegistration = () => {
+  waitingForRegistration.value = true;
+  clearInterval(waitingForRegistrationInterval);
+  waitingForRegistrationInterval = setInterval(loadOwnedEnsDomains, 10000);
```

Comment:
Sorry one more thing. This will show `RowLoading` every 10s after clicking on register, it doesn't look very nice when the UI bounces around like that I think. Can we do the `waitingForRegistrationInterval` without triggering that loading effect.

Reply:
Another good catch! :D I actually had that covered at first.... I gues I have "verschlimmbessert" it. ;)

Reply:
fixed :)

Reply:
Perfect :)